src/simulation/elements/PUMP.cpp

```diff
@@ -63,7 +63,7 @@ int Element_PUMP::update(UPDATE_FUNC_ARGS)
 
 		for (rx=-1; rx<2; rx++)
 			for (ry=-1; ry<2; ry++)
-				if (!(rx && ry))
+				if (parts[i].tmp == 0 && (!(rx && ry)))
```

Comment:
I also think there shouldn't be a .tmp = 0 restriction here. What I would do is, have a bool and another variable to keep track of whether or not some FILT was found, and then move this entire section below the other section. If the bool is set, use the predetermined pressure, else run the code inside this if